src/util/metricsbackend.h

```diff
@@ -122,17 +122,19 @@ class SimpleMetricGauge : public MetricGauge
 	double m_gauge;
 };
 
+using MetricLabels = std::map<std::string, std::string>;
+
 class MetricsBackend
 {
 public:
 	MetricsBackend() = default;
 
 	virtual ~MetricsBackend() {}
 
-	virtual MetricCounterPtr addCounter(
-			const std::string &name, const std::string &help_str);
-	virtual MetricGaugePtr addGauge(
-			const std::string &name, const std::string &help_str);
+	virtual MetricPtr addCounter(const std::string &name, const std::string &help_str,
```

Comment:
if you remove the Counter, it's implicit, and it's not implicit in any other prometheus API. Counter is not generic